src/OpenApi/test/Microsoft.AspNetCore.OpenApi.Tests/Services/OpenApiDocumentService/OpenApiDocumentServiceTests.Parameters.cs

```diff
@@ -190,4 +191,29 @@ await VerifyOpenApiDocument(builder, document =>
             Assert.Null(document.Paths["/api/content-type-lower"].Operations[OperationType.Get].Parameters);
         });
     }
+
+    [Fact]
+    public async Task GetOpenApiParameters_ToleratesCustomBindingSource()
+    {
+        var action = CreateActionDescriptor(nameof(ActionWithCustomBinder));
+
+        await VerifyOpenApiDocument(action, document =>
+        {
+            var operation = document.Paths["/custom-binding"].Operations[OperationType.Get];
+            var parameter = Assert.Single(operation.Parameters);
+            Assert.Equal("model", parameter.Name);
+            Assert.Null(parameter.In);
```

Comment:
Does this mean that a parameter is generated with a "name" but no "in" value? If so, that is an invalid OpenAPI definition, as "in" is a required field in a Parameter object.

https://github.com/OAI/OpenAPI-Specification/blob/3.0.3/versions/3.0.3.md#fixed-fields-10

If that's true, this makes me very uncomfortable.

Reply:
> If that's true, this makes me very uncomfortable.

Yep, this is correct. My thoughts on this here:

> One thing to note is that this means that users will have to do the due diligence to set the parameter location themselves via transformers since we can't figure out this information implicitly.

It's a bit icky that we don't set the default location ourselves but going from throwing an exception to setting null at least presents people with the option of telling us what ParameterLocation should be in a transformer instead of having to deal with an uncaught exception.

Reply:
Could we "just pick something", like "query", rather than leaving it empty? Alternatively, could we simply omit the parameter altogether? I think I would prefer either of these options to generating an invalid document.

Reply:
I like defaulting to query if that makes it easier for the transformer. Make it so (hope you like the reference).